napari/_app_model/_app.py

```diff
@@ -11,10 +11,32 @@
 from napari._app_model.actions._view_actions import VIEW_ACTIONS
 from napari._app_model.injection._processors import PROCESSORS
 from napari._app_model.injection._providers import PROVIDERS
+from napari.utils.key_bindings import _bind_plugin_key
 
 APP_NAME = 'napari'
 
 
+def populate_plugin_keymap():
+    """Populate the global plugin keymap from the app's keybinding registry."""
+    from napari._app_model import get_app
+
+    app = get_app()
+    for kb_rule in app.keybindings:
+        # skip built-in keybinds
+        if kb_rule.command_id.startswith('napari:'):
+            continue
+
+        # assign to local var so it doesn't get overwritten by future loops
+        # must come after continue statement to work yay python scoping
+        cmd_id = kb_rule.command_id
+
+        _bind_plugin_key(
+            kb_rule.keybinding,
+            lambda: app.commands.execute_command(cmd_id).result(),
+            overwrite=True,
+        )
```

Comment:
Based on my knowledge, reassigning the variable does not solve the problem of overwriting variable in the next loop. 

Using lambda expression in this context could cause problems with memory management. We decide not to use lambda for long-living binds. 

Also, I do not see here any prevention of overwriting existing shortcuts.